xhttp/xhttpserver/unmarshal_test.go

```diff
@@ -253,6 +256,13 @@ func testUnmarshalAnnotatedFull(t *testing.T) {
 						Header: http.Header{"X-Adhoc": []string{"adhoc value"}},
 					}.Then),
 				}.Annotated(),
+				func() *candlelight.Tracing {
```

Comment:
The catch is that that boolean simply means simply indicates whether the tracerProvider is the NoOp one (i.e. `Enabled=false  =>  TracerProvider is a NoOp`. 

I'm game for either: 
A) Renaming the variable to `IsNoop`
B) Drop the variable and let users inspect the TracerProvider and check if it's a noop by comparing it to `trace.NewNoopTracerProvider()`

Reply:
A big reason the concept of a `noop` is useful is to avoid `nil` checks.  So, just from first principles, I'd make sure client code wouldn't have to care about `candlelight.Tracing` being an optional component.  Here's a rough sketch of what I'm on about:

```
type Dependencies struct {
    fx.In
    Tracing candlelight.Tracing `optional:"true"` // this will be the zero value when no component is present
}

// in client code:
func setup(d Dependencies, h http.Handler) http.Handler {
    // client code can now always take the same codepath,
    // no conditionals necessary
    return d.Tracing.Next(h)
}
```